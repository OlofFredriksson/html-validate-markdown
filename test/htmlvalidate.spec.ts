import { HtmlValidate } from "html-validate";

const config = {
    extends: ["htmlvalidate:recommended"],
    transform: {
        "\\.(md)$": "<rootDir>",
    },
};

test('should find errors in "markdown.md"', () => {
    const htmlvalidate = new HtmlValidate(config);
    const report = htmlvalidate.validateFile("test/markdown.md");
    expect(report.valid).toBeTruthy();
    expect(report.results).toMatchSnapshot();
});

test('should find errors in "multiline-invalid.md"', () => {
    const htmlvalidate = new HtmlValidate(config);
    const report = htmlvalidate.validateFile("test/multiline-invalid.md");
    expect(report.valid).toBeFalsy();
    expect(report.results).toMatchSnapshot();
});
