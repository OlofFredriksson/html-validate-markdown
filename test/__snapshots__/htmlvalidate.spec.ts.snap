// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should find errors in "markdown.md" 1`] = `Array []`;

exports[`should find errors in "multiline-invalid.md" 1`] = `
Array [
  Object {
    "errorCount": 3,
    "filePath": "test/multiline-invalid.md",
    "messages": Array [
      Object {
        "column": 1,
        "line": 8,
        "message": "failed to tokenize \\"</div>\\\\n\\", expected attribute, \\">\\" or \\"/>\\".",
        "offset": 101,
        "ruleId": "parser-error",
        "selector": null,
        "severity": 2,
        "size": 0,
      },
      Object {
        "column": 6,
        "context": Object {
          "attribute": "type",
          "element": "button",
        },
        "line": 13,
        "message": "<button> is missing required \\"type\\" attribute",
        "offset": 12,
        "ruleId": "element-required-attributes",
        "selector": "div > button",
        "severity": 2,
        "size": 6,
      },
      Object {
        "column": 14,
        "context": undefined,
        "line": 13,
        "message": "Element <div> is not permitted as content in <button>",
        "offset": 20,
        "ruleId": "element-permitted-content",
        "selector": "div > button > div",
        "severity": 2,
        "size": 3,
      },
    ],
    "source": "# Markdown Header (Invalid Multiline html blocks)

\`\`\`html
<div something=\\"row\\">
    <button type=\\"invalid\\">
        Button not correctly clossed
    </button
</div>
\`\`\`

\`\`\`html
<div>
    <button><div>div in button</div></button>
</div>
\`\`\`
",
    "warningCount": 0,
  },
]
`;
